src/authorizations/components/redesigned/ResourceAccordionHeader.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {FC} from 'react'
+
+// Components
+import {
+  Accordion,
+  FlexBox,
+  AlignItems,
+  FlexDirection,
+  ComponentSize,
+  JustifyContent,
+  InputLabel,
+  Toggle,
+  InputToggleType,
+} from '@influxdata/clockface'
+
+interface OwnProps {
+  resourceName: string
+  permissions: any
+  onToggleAll: (name: string, permission: string, value: boolean) => void
+}
+
+export const ResourceAccordionHeader: FC<OwnProps> = props => {
+  const {resourceName} = props
+  const readToggle = () => {
+    const {permissions} = props
+    let tag = true
+    if (permissions && Object.keys(permissions).length > 0) {
+      Object.keys(permissions).map(key => {
+        permissions[key].permissions.read === true
```

Comment:
if I understand this correctly - we're trying to get the value of all permissions in the accordion and if they're all `true`, set the header as `true`, right?

i think `some` is the most appropriate function here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some.

>`some` tests whether at least one element in the array passes the test implemented by the provided function. It returns true if, in the array, it finds an element for which the provided function returns true; otherwise it returns false. It doesn't modify the array. 

```ts
tag = !Object.keys(permissions).some(key => permissions[key].permissions.read === false)
```

if it finds one instance of `read` equal to false, `some` passes the check and returns `true`. we negate the call (`!Object.keys...`), so if `false` is found, we'll actually set the value of `tag` to false and vice versa if it's not found.